diff --git a/activity.py b/activity.py
index eab2174..0963725 100644
--- a/activity.py
+++ b/activity.py
@@ -55,6 +55,8 @@ from sugar3.activity.widgets import StopButton
 
 from videos import EXERCISES, DANCES
 
+from sugarapp.widgets import SugarCompatibleActivity
+
 Gst.init(None)
 
 
@@ -250,9 +252,9 @@ class VideoButton(Gtk.EventBox):
         # self.window.set_cursor(Gdk.Cursor.new(Gdk.HAND2))
         self.get_window().set_cursor(Gdk.Cursor(Gdk.CursorType.HAND2))
 
-class SwiftFeetActivity(activity.Activity):
+class SwiftFeetActivity(SugarCompatibleActivity):
     def __init__(self, handle):
-        activity.Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
         self._current_video_idx = None
         self.max_participants = 1
 
